src/resources/movies.ts

```diff
@@ -0,0 +1,254 @@
+    try {
+        await db.collection('movies').insertOne(movie);
+        const movies = await db.collection('movies').find().toArray();
+        res.status(201).json(movies);
+    } catch (error) {
+        console.error(error);
+        res.status(500).json({ error: "Internal server error." });
+    }
+  });
+  
+  /**
+   * @swagger
+   * /movies/{title}:
+   *   put:
+   *     summary: Update a movie by title
+   *     description: Update a movie in the database by its title.
+   *     parameters:
+   *       - in: path
+   *         name: title
+   *         required: true
+   *         description: The title of the movie to update.
+   *         schema:
+   *           type: string
+   *     requestBody:
+   *       description: New movie data to update.
+   *       required: true
+   *       content:
+   *         application/json:
+   *           schema:
+   *             type: object
+   *             properties:
+   *               title:
+   *                 type: string
+   *                 example: The Matrix
+   *               genre:
+   *                 type: array
+   *                 items:
+   *                   type: string
+   *                 example: ["Action", "Adventure", "Science Fiction"]
+   *               releaseDate:
+   *                 type: string
+   *                 example: 1999-03-31
+   *               description:
+   *                 type: string
+   *                 example: A classic sci-fi movie.
+   *     responses:
+   *       200:
+   *         description: Movie updated successfully.
+   *       400:
+   *         description: Bad request - Invalid data provided.
+   *       404:
+   *         description: Movie not found.
+   */
+  updateMovieByTitleRouter.put('/:title', async (req: Request, res: Response): Promise<void> => {
+    const title = req.params.title;
+    const movie = await db.collection('movies').findOne({ title: title });
+  
+    if (!movie) {
+        res.status(404).json({ error: "Movie not found." });
+        return;
+    }
+  
+    const editedMovie = {
+        "title": title + '1',
+        "genre": movie.genre,
+        "releaseDate": movie.releaseDate,
+        "description": movie.description
+    };
+  
+    try {
+        await db.collection('movies').updateOne({ title: title }, { $set: editedMovie });
+        const movieCollection = await db.collection('movies').find().toArray();
+        res.status(200).json(movieCollection);
+    } catch (error) {
+        console.error(error);
+        res.status(500).json({ error: "Internal server error." });
+    }
+  });
+  
+  /**
+   * @swagger
+   * /movies/{id}:
+   *   delete:
+   *     summary: Delete a movie by ID
+   *     description: Delete a movie from the database by its ID.
+   *     parameters:
+   *       - in: path
+   *         name: id
+   *         required: true
+   *         description: The ID of the movie to delete.
+   *         schema:
+   *           type: string
+   *           format: uuid
```

Comment:
Also, the endpoint description specifies exactly this `uuid`, so the ObjectId won't work in the swagger docs.

![image](https://github.com/MaxSihua/Node.js_Course_Practice/assets/113895351/dba0ef41-b3d9-46b6-9c6c-dbe9589e374f)

Reply:
As a solution you can describe separate field id and describe it's type like uuid or replace with proper type.

Reply:
I roughly understand what needs to be done, but I don't understand how to do it.

Reply:
Maybe I just didn't quite understand the problem and there is an opportunity to write it in a little more detail. Thank you

Reply:
The problem is that Swagger waiting for uuid entity but for proper deletion of entity there is a need to provide _id with type ObjectId. Uuid type is not the same thing as MongoDB ObjectId type. You need to raplace uuid type with proper one. The tip is under screenshot :)

Reply:
done